src/ln/channel.rs

```diff
@@ -2232,6 +2344,14 @@ impl Channel {
 				log_debug!(self, "Reconnected channel {} with only lost remote commitment tx", log_bytes!(self.channel_id()));
 			}
 
+			if self.channel_state & (ChannelState::MonitorUpdateFailed as u32) != 0 {
+				debug_assert!(order == RAACommitmentOrder::RevokeAndACKFirst);
+				self.monitor_pending_commitment_signed = true;
```

Comment:
Order is RevokeAndACKFirst here but we require sending commitment_signed first ?

Reply:
Nice catch! Found a hole in my tests, too, so added a new one which hits this block.